### ISSUE TITLE
How to?

### ISSUE THREAD

**Issue by askfriends** [pownthep/Vibe#8]
Please add some info how to make it work with out google account/team drive?

there is no info provided for noobs.

**Comment by pownthep** [comment]
Ideally, I want to make the app work without forcing the user to sign into their google account. But the app depends on being able to use Google Drive REST API which requires an access token which you can only get if you have a google account.

**Comment by askfriends** [comment]
> Ideally, I want to make the app work without forcing the user to sign into their google account. But the app depends on being able to use Google Drive REST API which requires an access token which you can only get if you have a google account.

you mean users wont need to login using google account?
thats a great feature if you could manage to add btw..

but i want to know can we make server like yours using our team drive or shared drive on web version?

**Comment by pownthep** [comment]
> > Ideally, I want to make the app work without forcing the user to sign into their google account. But the app depends on being able to use Google Drive REST API which requires an access token which you can only get if you have a google account.
> 
> you mean users wont need to login using google account?
> thats a great feature if you could manage to add btw..
> 
> but i want to know can we make server like yours using our team drive or shared drive on web version?

Yes, you can definitely use your own drive or shared drive to host your files. Here's what you need:
1. An access token associated with a Google Account
2. The shared drive id 

Then, you get a list of files in your shared drive by:
```javascript
const url = "https://www.googleapis.com/drive/v3/files?driveId=YOUR_SHARED_DRIVE_ID";

const response = await axios({
        url,
        method: "GET",
        headers: {
          Authorization: "Bearer " + ACCESS_TOKEN,
          Accept: application/json
        },
});

const arrayOfFiles = response.data 
```

As for streaming the video file, there is two option that I know, 
1. Run a localhost server to download video data and forward data to the video player - Requires sign in and does not work on browser
2. Use an API KEY - Does not require sign in and works on browser. So, your video source URL would be
```bash
https://www.googleapis.com/drive/v3/files/FILE_ID?alt=media&key=[YOUR_API_KEY]
```
You can learn how to create an API key in this article - https://developers.google.com/maps/documentation/javascript/get-api-key

Keep in mind that browsers only supports h.264 video with AAC audio format.

**Comment by askfriends** [comment]
> > > Ideally, I want to make the app work without forcing the user to sign into their google account. But the app depends on being able to use Google Drive REST API which requires an access token which you can only get if you have a google account.
> > 
> > 
> > you mean users wont need to login using google account?
> > thats a great feature if you could manage to add btw..
> > but i want to know can we make server like yours using our team drive or shared drive on web version?
> 
> Yes, you can definitely use your own drive or shared drive to host your files. Here's what you need:
> 
> 1. An access token associated with a Google Account
> 2. The shared drive id
> 
> Then, you get a list of files in your shared drive by:
> 
> ```js
> const url = "https://www.googleapis.com/drive/v3/files?driveId=YOUR_SHARED_DRIVE_ID";
> 
> const response = await axios({
>         url,
>         method: "GET",
>         headers: {
>           Authorization: "Bearer " + ACCESS_TOKEN,
>           Accept: application/json
>         },
> });
> 
> const arrayOfFiles = response.data 
> ```
> 
> As for streaming the video file, there is two option that I know,
> 
> 1. Run a localhost server to download video data and forward data to the video player - Requires sign in and does not work on browser
> 2. Use an API KEY - Does not require sign in and works on browser. So, your video source URL would be
> 
> ```shell
> https://www.googleapis.com/drive/v3/files/FILE_ID?alt=media&key=[YOUR_API_KEY]
> ```
> 
> You can learn how to create an API key in this article - https://developers.google.com/maps/documentation/javascript/get-api-key
> 
> Keep in mind that browsers only supports h.264 video with AAC audio format.

Thanks for detailed reply.
but i am in pro in coding, so will you implement that code in your main repo?
or create another repo and make it specific for using our team drives?
i dont know where to use that code you have specified.

**Comment by ssnjrthegr8** [comment]
This project looks quite interesting but it seems like that it is not completed? 

Also how does the mpv.js aspect of this project work? Can I make my browser play any video I want through mpv?
